### PR TITLE
Use in-tree cabal in build.cabal.sh

### LINKED ISSUE #274: build.cabal.sh is broken
There seems to be some kind of mismatch with `Cabal` library versions. From a fresh clean,

```
~/Documents/haskell/ghc-hadrian:ghc/hadrian/build.cabal.sh --flavour=quickest -j4 
In order, the following will be built (use -v for more details):
shake-0.15.10
hadrian-0.1.0.0
Configuring shake-0.15.10...
Building shake-0.15.10...
Configuring hadrian-0.1.0.0...
Preprocessing executable 'hadrian' for hadrian-0.1.0.0...
[ 1 of 93] Compiling Stage            ( src/Stage.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Stage.o ) [Development.Shake.Classes changed]
[ 2 of 93] Compiling CmdLineFlag      ( src/CmdLineFlag.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/CmdLineFlag.o ) [Data.List.Extra changed]
[ 3 of 93] Compiling Base             ( src/Base.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Base.o ) [Development.Shake.FilePath changed]
[ 4 of 93] Compiling Environment      ( src/Environment.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Environment.o ) [Base changed]
[ 5 of 93] Compiling Oracles.PackageData ( src/Oracles/PackageData.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Oracles/PackageData.o ) [Development.Shake.Config changed]
[ 6 of 93] Compiling Package          ( src/Package.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Package.o ) [Base changed]
[ 7 of 93] Compiling Oracles.Config   ( src/Oracles/Config.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Oracles/Config.o ) [Development.Shake.Config changed]
[ 8 of 93] Compiling Oracles.Config.Setting ( src/Oracles/Config/Setting.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Oracles/Config/Setting.o ) [Base changed]
[ 9 of 93] Compiling Way              ( src/Way.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Way.o ) [Base changed]
[10 of 93] Compiling Context          ( src/Context.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Context.o ) [Base changed]
[11 of 93] Compiling Oracles.LookupInPath ( src/Oracles/LookupInPath.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Oracles/LookupInPath.o ) [Base changed]
[12 of 93] Compiling Oracles.WindowsPath ( src/Oracles/WindowsPath.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Oracles/WindowsPath.o ) [Base changed]
[13 of 93] Compiling Oracles.Config.Flag ( src/Oracles/Config/Flag.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Oracles/Config/Flag.o ) [Base changed]
[14 of 93] Compiling Oracles.DirectoryContent ( src/Oracles/DirectoryContent.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Oracles/DirectoryContent.o ) [System.Directory.Extra changed]
[15 of 93] Compiling GHC              ( src/GHC.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/GHC.o ) [Base changed]
[16 of 93] Compiling Builder          ( src/Builder.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Builder.o ) [Base changed]
[17 of 93] Compiling Target           ( src/Target.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Target.o ) [Base changed]
[18 of 93] Compiling Expression       ( src/Expression.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Expression.o ) [Base changed]
[19 of 93] Compiling Flavour          ( src/Flavour.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Flavour.o ) [Expression changed]
[20 of 93] Compiling Predicate        ( src/Predicate.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Predicate.o ) [Base changed]
[21 of 93] Compiling Settings.Builders.Alex ( src/Settings/Builders/Alex.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Builders/Alex.o ) [Expression changed]
[22 of 93] Compiling Settings.Builders.Ar ( src/Settings/Builders/Ar.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Builders/Ar.o ) [Base changed]
[23 of 93] Compiling Settings.Builders.GenApply ( src/Settings/Builders/GenApply.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Builders/GenApply.o ) [Expression changed]
[24 of 93] Compiling Settings.Builders.GenPrimopCode ( src/Settings/Builders/GenPrimopCode.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Builders/GenPrimopCode.o ) [Expression changed]
[25 of 93] Compiling Settings.Builders.Happy ( src/Settings/Builders/Happy.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Builders/Happy.o ) [Expression changed]
[26 of 93] Compiling Settings.Builders.Ld ( src/Settings/Builders/Ld.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Builders/Ld.o ) [Expression changed]
[27 of 93] Compiling Settings.Builders.Tar ( src/Settings/Builders/Tar.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Builders/Tar.o ) [Expression changed]
[28 of 93] Compiling Rules.Generators.Common ( src/Rules/Generators/Common.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Rules/Generators/Common.o ) [Base changed]
[29 of 93] Compiling Rules.Generators.GhcAutoconfH ( src/Rules/Generators/GhcAutoconfH.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Rules/Generators/GhcAutoconfH.o ) [Base changed]
[30 of 93] Compiling Rules.Generators.GhcBootPlatformH ( src/Rules/Generators/GhcBootPlatformH.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Rules/Generators/GhcBootPlatformH.o ) [Expression changed]
[31 of 93] Compiling Rules.Generators.GhcPlatformH ( src/Rules/Generators/GhcPlatformH.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Rules/Generators/GhcPlatformH.o ) [Expression changed]
[32 of 93] Compiling Rules.Generators.GhcSplit ( src/Rules/Generators/GhcSplit.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Rules/Generators/GhcSplit.o ) [Base changed]
[33 of 93] Compiling Rules.Generators.GhcVersionH ( src/Rules/Generators/GhcVersionH.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Rules/Generators/GhcVersionH.o ) [Expression changed]
[34 of 93] Compiling Rules.Generators.VersionHs ( src/Rules/Generators/VersionHs.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Rules/Generators/VersionHs.o ) [Expression changed]
[35 of 93] Compiling Rules.Wrappers.Ghc ( src/Rules/Wrappers/Ghc.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Rules/Wrappers/Ghc.o ) [Base changed]
[36 of 93] Compiling Settings.Packages.Directory ( src/Settings/Packages/Directory.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Packages/Directory.o ) [Expression changed]
[37 of 93] Compiling Settings.Packages.GhcPrim ( src/Settings/Packages/GhcPrim.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Packages/GhcPrim.o ) [Expression changed]
[38 of 93] Compiling Settings.Packages.Haddock ( src/Settings/Packages/Haddock.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Packages/Haddock.o ) [Expression changed]
[39 of 93] Compiling Settings.Packages.IservBin ( src/Settings/Packages/IservBin.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Packages/IservBin.o ) [Expression changed]
[40 of 93] Compiling Settings.Packages.RunGhc ( src/Settings/Packages/RunGhc.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Packages/RunGhc.o ) [Expression changed]
[41 of 93] Compiling UserSettings     ( src/UserSettings.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/UserSettings.o )

src/UserSettings.hs:20:1: warning: [-Wunused-top-binds]
    Defined but not used: ‘userArgs’
[42 of 93] Compiling Settings.Paths   ( src/Settings/Paths.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Paths.o ) [Base changed]
[43 of 93] Compiling Settings.Default[boot] ( src/Settings/Default.hs-boot, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Default.o-boot )
[44 of 93] Compiling Settings.Flavours.Quickest ( src/Settings/Flavours/Quickest.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Flavours/Quickest.o ) [Expression changed]
[45 of 93] Compiling Settings.Builders.Configure ( src/Settings/Builders/Configure.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Builders/Configure.o ) [Base changed]
[46 of 93] Compiling Settings.Builders.Make ( src/Settings/Builders/Make.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Builders/Make.o ) [Base changed]
[47 of 93] Compiling Oracles.ModuleFiles ( src/Oracles/ModuleFiles.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Oracles/ModuleFiles.o ) [Base changed]
[48 of 93] Compiling Rules.Selftest   ( src/Rules/Selftest.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Rules/Selftest.o ) [Development.Shake changed]
[49 of 93] Compiling Rules.Wrappers.GhcPkg ( src/Rules/Wrappers/GhcPkg.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Rules/Wrappers/GhcPkg.o ) [Base changed]
[50 of 93] Compiling Settings.Flavours.Quick ( src/Settings/Flavours/Quick.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Flavours/Quick.o ) [Context changed]
[51 of 93] Compiling Settings         ( src/Settings.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings.o ) [Base changed]
[52 of 93] Compiling Settings.Builders.Common ( src/Settings/Builders/Common.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Builders/Common.o ) [Base changed]
[53 of 93] Compiling Settings.Builders.DeriveConstants ( src/Settings/Builders/DeriveConstants.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Builders/DeriveConstants.o ) [Base changed]
[54 of 93] Compiling Settings.Builders.Cc ( src/Settings/Builders/Cc.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Builders/Cc.o ) [Base changed]
[55 of 93] Compiling Settings.Builders.Hsc2Hs ( src/Settings/Builders/Hsc2Hs.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Builders/Hsc2Hs.o ) [Base changed]
[56 of 93] Compiling Oracles.ArgsHash ( src/Oracles/ArgsHash.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Oracles/ArgsHash.o ) [Base changed]
[57 of 93] Compiling Rules.Actions    ( src/Rules/Actions.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Rules/Actions.o ) [System.Directory.Extra changed]
[58 of 93] Compiling Rules.Test       ( src/Rules/Test.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Rules/Test.o ) [Base changed]
[59 of 93] Compiling Rules.Perl       ( src/Rules/Perl.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Rules/Perl.o ) [Base changed]
[60 of 93] Compiling Settings.Builders.GhcCabal ( src/Settings/Builders/GhcCabal.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Builders/GhcCabal.o ) [Base changed]
[61 of 93] Compiling Settings.Builders.Ghc ( src/Settings/Builders/Ghc.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Builders/Ghc.o ) [Base changed]
[62 of 93] Compiling Settings.Builders.GhcPkg ( src/Settings/Builders/GhcPkg.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Builders/GhcPkg.o ) [Base changed]
[63 of 93] Compiling Oracles.Dependencies ( src/Oracles/Dependencies.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Oracles/Dependencies.o ) [Base changed]
[64 of 93] Compiling Rules.Compile    ( src/Rules/Compile.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Rules/Compile.o ) [Base changed]
[65 of 93] Compiling Settings.Builders.Haddock ( src/Settings/Builders/Haddock.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Builders/Haddock.o ) [Base changed]
[66 of 93] Compiling Settings.Builders.HsCpp ( src/Settings/Builders/HsCpp.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Builders/HsCpp.o ) [Expression changed]
[67 of 93] Compiling Settings.Packages.Base ( src/Settings/Packages/Base.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Packages/Base.o ) [Base changed]
[68 of 93] Compiling Settings.Packages.Compiler ( src/Settings/Packages/Compiler.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Packages/Compiler.o ) [Base changed]
[69 of 93] Compiling Settings.Packages.Ghc ( src/Settings/Packages/Ghc.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Packages/Ghc.o ) [Expression changed]
[70 of 93] Compiling Settings.Packages.GhcCabal ( src/Settings/Packages/GhcCabal.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Packages/GhcCabal.o ) [Base changed]
[71 of 93] Compiling Settings.Packages.Hp2ps ( src/Settings/Packages/Hp2ps.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Packages/Hp2ps.o ) [Base changed]
[72 of 93] Compiling Settings.Packages.IntegerGmp ( src/Settings/Packages/IntegerGmp.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Packages/IntegerGmp.o ) [Base changed]
[73 of 93] Compiling Settings.Packages.Rts ( src/Settings/Packages/Rts.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Packages/Rts.o ) [Base changed]
[74 of 93] Compiling Settings.Packages.Touchy ( src/Settings/Packages/Touchy.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Packages/Touchy.o ) [Base changed]
[75 of 93] Compiling Settings.Packages.Unlit ( src/Settings/Packages/Unlit.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Packages/Unlit.o ) [Base changed]
[76 of 93] Compiling Settings.Default ( src/Settings/Default.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Settings/Default.o ) [Base changed]
[77 of 93] Compiling Oracles.PackageDatabase ( src/Oracles/PackageDatabase.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Oracles/PackageDatabase.o ) [Base changed]
[78 of 93] Compiling Rules.Oracles    ( src/Rules/Oracles.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Rules/Oracles.o ) [Base changed]
[79 of 93] Compiling Rules.Generators.ConfigHs ( src/Rules/Generators/ConfigHs.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Rules/Generators/ConfigHs.o ) [Base changed]
[80 of 93] Compiling Rules.Libffi     ( src/Rules/Libffi.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Rules/Libffi.o ) [Base changed]
[81 of 93] Compiling Rules.Generate   ( src/Rules/Generate.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Rules/Generate.o ) [Base changed]
[82 of 93] Compiling Rules.Clean      ( src/Rules/Clean.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Rules/Clean.o ) [Base changed]
[83 of 93] Compiling Rules.Data       ( src/Rules/Data.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Rules/Data.o ) [Base changed]
[84 of 93] Compiling Rules.Dependencies ( src/Rules/Dependencies.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Rules/Dependencies.o )
[85 of 93] Compiling Rules.Documentation ( src/Rules/Documentation.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Rules/Documentation.o ) [Base changed]
[86 of 93] Compiling Rules.Cabal      ( src/Rules/Cabal.hs, /Users/matt/Documents/haskell/ghc-hadrian/ghc/hadrian/dist-newstyle/build/hadrian-0.1.0.0/build/hadrian/hadrian-tmp/Rules/Cabal.o )

src/Rules/Cabal.hs:39:71: error:
    • Variable not in scope:
        condLibraries
          :: GenericPackageDescription -> [(a1, CondTree v0 [Dependency] a0)]
    • Perhaps you meant ‘condLibrary’ (imported from Distribution.PackageDescription)
```

### PR DESCRIPTION
**Pull request by KaiHa**
Fix #274

### CODE REVIEW

**Comment by mpickering** [comment]
As you note, this works for < 1.24 but I am using Cabal 1.24 where this error is triggered.

**Comment by KaiHa** [comment]
@mpickering Can you test my latest change? It has worked with cabal 1.24 at least in my setup.

File: build.cabal.sh [hunk 1]
```diff
@@ -61,6 +61,11 @@ else
     # Initialize sandbox if necessary
     if ! ( "$CABAL" sandbox hc-pkg list > /dev/null 2>&1); then
         "$CABAL" sandbox init
+        ( cd ../libraries/Cabal/Cabal
```
Comment (mpickering) [inline build.cabal.sh]: Why not use `cabal sandbox add-source`?

**Comment by mpickering** [comment]
@KaiHa That is how I was going to fix it but you beat me to it! However, I don't think the pre-1.24 fix is really right. See my comment.

File: build.cabal.sh (hunk 1 above)
Comment (KaiHa) [inline build.cabal.sh]: > Why not use cabal sandbox add-source?

To be honest, I never heard of `add-source` before. I will have a look at it later this day.

Comment (mpickering) [inline build.cabal.sh]: I think you just need to replace this section with 

`cabal sandbox add-source ../libraries/Cabal/Cabal`

**Comment by mpickering** [comment]
The project file fixes the build for me.

**Comment by snowleopard** [comment]
Many thanks @KaiHa and @mpickering. Merged!

**Comment by hvr** [comment]
@snowleopard just so I understand this; this is only for the legacy cabal support? I.e. do I need to fixup  anything for (recommended) `cabal new-build` mode ? Just asking, because I haven't tried Hadrian for weeks...

**Comment by snowleopard** [comment]
@hvr My understanding is that this fix will help both `cabal build` and `cabal new-build`, as both will need to have the right paths to the in-tree Cabal library `../libraries/Cabal/Cabal`. 

Perhaps, you could test that this also works correctly with `cabal new-build`?

**Comment by mpickering** [comment]
I tested the new-build path. It was fixed by adding the `cabal.project` file. No other changes were necessary.

**Comment by snowleopard** [comment]
@mpickering :+1:

**Pull request merged**